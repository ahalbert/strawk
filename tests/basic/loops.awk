BEGIN {
  a = 0
  while (a < 3) {
    print "a"
    a++
  }
  b = 0
  do {
    print "b"
  } while (b < 0)

  for (i = 0; i < 3; ++i) {
   print i 
  }

  for (i = 0; i < 3; ++i) {
   print "c"
   break
  }

  for (i = 0; i < 3; ++i) {
   print "d"
   continue
   print "e"
  }

  arr[0] = 1
  arr[1] = 1
  arr["a"] = 1
  for (k in arr) {
  print k, arr[k]
  }
}
