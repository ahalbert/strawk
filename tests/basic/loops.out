a
a
a
b
0
1
2
c
d
d
d
0 1
1 1
a 1
